Bio/Seq.py

```diff
@@ -1280,6 +1281,53 @@ def ungap(self, gap=None):
             raise ValueError("Unexpected gap character, {0!r}".format(gap))
         return Seq(str(self).replace(gap, ""), alpha)
 
+    def join(self, other):
+        """Return a merge of the sequences in other, spaced by the sequence from self.
+
+        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+
+        >>> concatenated = Seq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable
+        are not Seq or String objects
+        """
+        if not isinstance(other, collections.Iterable):  # doesn't detect single strings
+            raise ValueError("Input must be an iterable")
+        if isinstance(other, basestring):
+            raise ValueError("Input must be an iterable")
+        from Bio.SeqRecord import SeqRecord  # Lazy to avoid circular imports
+        temp_data = ""
+        # if the spacer is empty it will initially default none
+        if self._data == "":
+            a = None
+        else:
+            a = self.alphabet
+        for c in other:
+            if isinstance(c, SeqRecord):
+                raise TypeError("Iterable cannot contain SeqRecords")
+            elif hasattr(c, "alphabet"):
+                if a is None:  # if spacer is empty alphabet defaults to type of the first sequence
+                    a = c.alphabet
+                else:
+                    if not Alphabet._check_type_compatible([a, c.alphabet]):
+                        raise TypeError(
+                            "Incompatible alphabets {0!r} and {1!r}".format(
+                                a, c.alphabet))
+                    a = Alphabet._consensus_alphabet([a, c.alphabet])
+                temp_data += str(self) + str(c)
+            elif isinstance(c, basestring):
+                temp_data += str(self) + str(c)
+            else:
+                raise ValueError("Input must be an iterable of Seqs or Strings")
+
```

Comment:
If you remove line 1319, replace 1321 with ```pass``` and add below 1323:

```python
temp_data = str(self).join([str(z) for z in other])
```
you can gain another 1.6x in speed.

Reply:
Good - I did think that would be quicker, but it was masked by the seq/alphabet overhead until your suggestion for the common case of everything having the same alphabet object.

Reply:
You're correct, I previously misunderstood and tested this idea poorly. I've updated it to perform this join.